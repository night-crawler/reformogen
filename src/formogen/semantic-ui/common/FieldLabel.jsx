import React from 'react';
import PropTypes from 'prop-types';
import { Popup } from 'semantic-ui-react';
import { upperFirst } from 'lodash';

FieldLabel.displayName = 'FieldLabel';
FieldLabel.propTypes = {
  upperFirstLabel: PropTypes.bool,
  verbose_name: PropTypes.string.isRequired,
  help_text: PropTypes.string,
  /** show help text on hover label (?) sign, or put in the bottom */
  helpTextOnHover: PropTypes.bool,
};
export function FieldLabel(props) {
  const labelText = props.upperFirstLabel 
    ? upperFirst(props.verbose_name) 
    : props.verbose_name;

  if (!props.help_text || !props.helpTextOnHover)
    return <label>{ labelText }</label>;

  /* replaced trigger's <Icon name="help" /> with <i> to prevent react warning:
     * Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.
  */
  return (
    <label>
      <Popup
        trigger={ <i className='icon help' /> }
        content={ props.help_text }
        wide={ true }
        position='bottom left'
      />
      { labelText }
    </label>
  );
}
