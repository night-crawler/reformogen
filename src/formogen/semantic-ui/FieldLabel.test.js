import { FieldLabel } from './FieldLabel';


it('<FieldLabel />', () => {
  const wrapper = shallow(
    <FieldLabel
      type='Label'
      name='Label: name'
      verbose_name='Label: verbose_name'
      help_text='Label: help_text'
      layoutOpts={ { width: 1 } }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
