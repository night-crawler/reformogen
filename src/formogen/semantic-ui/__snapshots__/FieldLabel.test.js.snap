// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FieldLabel
    help_text="Label: help_text"
    layoutOpts={
      Object {
        "width": 1,
      }
    }
    name="Label: name"
    type="Label"
    verbose_name="Label: verbose_name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Label: verbose_name",
    },
    "ref": null,
    "rendered": "Label: verbose_name",
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Label: verbose_name",
      },
      "ref": null,
      "rendered": "Label: verbose_name",
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
